examples/AcousticWave2D_codegen.py

```diff
@@ -95,21 +95,43 @@ def _init_taylor(self, nt):
         # domain to the border
 
         # Forward wave equation
+
+        def damp_boundary(damp):
+            h = self.h
+            dampcoeff = 1.5 * np.log(1.0 / 0.001) / (40 * h)
+            nx, ny = self.model.get_shape()
+            nbpml = self.nbpml
+            for i in range(nbpml):
+                pos = np.abs((nbpml-i)/nbpml)
+                val = dampcoeff * (pos - np.sin(2*np.pi*pos)/(2*np.pi))
+                damp[i, :] += val
+                damp[-i, :] += val
+                damp[:, i] += val
+                damp[:, -i] += val
+
         wave_equation = m * dtt - (dxx + dyy) + e * dt
         stencil = solve(wave_equation, p(x, y, t+s))[0]
         self.nt = nt
-
+        m_sub = DenseData("m", self.model.vp.shape, self.dtype)
+        m_sub.initializer = lambda ref: np.copyto(ref, self.model.vp**(-2))
```

Comment:
I was basing it on the fact that the numpy manual repeatedly mentions that numpy tries to reduce copying data unless it needs to. If I understand correctly, the page [here](http://docs.scipy.org/doc/numpy/reference/arrays.indexing.html) says that even slicing usually returns a view. 
Also, if I understand this, we need to minimize copying of data in our code as well; especially since these data structures will become huge very quickly. So I was assuming that ``initialize_model_vp`` is the way to go even in the near future. I implemented a basic version of that method in my initializer that makes a copy so as to minimise upstream changes for now. This would eventually become such that the only time memory is allocated for the model is in our code and upstream code only populates the data structures we create. I thought this is what we discussed when we had the discussion over the API.
As of now I am trying to make the interface look as similar to what we discussed as I can without making major changes in the upstream code. The upstream changes would have to be made soon too but I am just going one step at a time for now.